rocketmq-console/src/main/java/org/apache/rocketmq/console/util/MsgTraceDecodeUtil.java

```diff
@@ -88,9 +118,24 @@
                 subAfterContext.getTraceBeans().add(bean);
                 subAfterContext.setCostTime(Integer.parseInt(line[3]));
                 subAfterContext.setSuccess(Boolean.parseBoolean(line[4]));
-                if (line.length >= 7) {
-                    // add the context type
-                    subAfterContext.setContextCode(Integer.parseInt(line[6]));
+                // compatible with different version
+                switch (line.length) {
+                    case TRACE_MSG_SUBAFTER_V1_LEN:
+                        break;
+                    case TRACE_MSG_SUBAFTER_V2_LEN:
+                        subAfterContext.setContextCode(Integer.parseInt(line[6]));
+                        break;
+                    case TRACE_MSG_SUBAFTER_V3_LEN:
+                        subAfterContext.setContextCode(Integer.parseInt(line[6]));
+                        subAfterContext.setTimeStamp(Long.parseLong(line[7]));
+                        subAfterContext.setGroupName(line[8]);
+                        break;
+                    default:
```

Comment:
What's the difference between TRACE_MSG_SUBAFTER_V1_LEN and default logic, It is correct here?

Reply:
default logic is compatible for new version subAfterType.
If a new field is added to subAfter message, the length will greater than TRACE_MSG_SUBAFTER_V3_LEN.